posts/gsoc-2020-blog-3.md

```diff
@@ -0,0 +1,219 @@
+
+```pycon
+>>> import cantera as ct                           #import Cantera's Python module
+>>> gas = ct.Solution('gri30.cti')                 #create a default GRI-Mech 3.0 gas mixture
+>>> gas.TPX = 1000.0, ct.one_atm, 'H2:2,O2:1,N2:4' #set gas to an interesting state
+>>> reac = ct.IdealGasReactor(gas)                 #create a reactor containing the gas
+>>> sim = ct.ReactorNet([reac])                    #add the reactor to a new ReactorNet simulator
+>>> gas()                #view the initial state of the mixture (state summary will be printed to console)
+>>> sim.advance(1)       #advance the simulation to the specified absolute time, t = 1 sec
+>>> gas()                #view the updated state of the mixture, reflecting properties at t = 1 sec
+```
+
+Equivalently, the following can be compiled and run using Cantera's C++ interface:
+
+```c++
+#include "cantera/zerodim.h" //include Cantera's 0D reactor simulation module
+using namespace Cantera;     //activate Cantera namespace to identify scope of class and method references
+int main() {
+    auto gas = newSolution("gri30.cti");  //create a default GRI-Mech 3.0 gas mixture
+    gas->thermo()->setState_TPX(1000.0, OneAtm, "H2:2,O2:1,N2:4"); //set gas to an interesting state
+    Reactor reac;                         //create an empty Reactor
+    reac.insert(gas);                     //fill the reactor with the specified gas
+    ReactorNet sim;                       //create an empty ReactorNet simulator
+    sim.addReactor(reac);                 //add the reactor to the ReactorNet
+    std::cout << gas->thermo()->report(); //print the intitial state of the mixture to the console
+    sim.advance(1);                       //advance the simulation to absolute time t = 1 sec
+    std::cout << gas->thermo()->report(); //print the updated state of the mixture to the console
+    return 0;
+}
+```
+
+For a more advanced example that adds inlets and outlets to the reactor, see Cantera's combustor example ([Python](https://github.com/Cantera/cantera/blob/main/interfaces/cython/cantera/examples/reactors/combustor.py) | [C++](https://github.com/Cantera/cantera/blob/main/samples/cxx/combustor/combustor.cpp)). Additional examples can be found in the [Python Reactor Network Examples](https://cantera.org/examples/python/index.html#python-example-reactors) section of the Cantera website. In any case, Cantera performs reactor network time integration by generating an ODE system that represents the network, and then solving this system via an external `Integrator` object. What is an `Integrator`?
```

Comment:
I think "In any case" could start a new paragraph and explicitly call out that `advance()` is the step you're going to talk about here